heron/schedulers/tests/java/com/twitter/heron/scheduler/yarn/YarnLauncherTest.java

```diff
@@ -0,0 +1,101 @@
+import com.twitter.heron.scheduler.yarn.HeronConfigurationOptions.Role;
+import com.twitter.heron.scheduler.yarn.HeronConfigurationOptions.TopologyJar;
+import com.twitter.heron.scheduler.yarn.HeronConfigurationOptions.TopologyName;
+import com.twitter.heron.scheduler.yarn.HeronConfigurationOptions.TopologyPackageName;
+import com.twitter.heron.spi.common.Config;
+import com.twitter.heron.spi.common.Keys;
+
+public class YarnLauncherTest {
+  private Map<String, String> testConfigMap = new HashMap<>();
+  private Map<String, String> expectedMap = new HashMap<>();
+
+  @Test
+  public void getHMDriverConfConstructsReefConfig() throws Exception {
+    YarnLauncher launcher = new YarnLauncher();
+    YarnLauncher spyLauncher = Mockito.spy(launcher);
+    Mockito.doNothing().when(spyLauncher).addLibraryToClasspathSet(Mockito.anyString());
+
+    testConfigMap.clear();
+    expectedMap.clear();
+    setConfigs(Keys.topologyName(), "topology", TopologyName.class);
+    setConfigs(Keys.topologyBinaryFile(), "binary", TopologyJar.class);
+    setConfigs(Keys.topologyPackageFile(), "package", TopologyPackageName.class);
+    setConfigs(Keys.cluster(), "cluster", Cluster.class);
+    setConfigs(Keys.role(), "role", Role.class);
+    setConfigs(Keys.environ(), "env", Environ.class);
+    setConfigs(YarnContext.HERON_SCHEDULER_YARN_QUEUE, "queue", JobQueue.class);
+    setConfigs(YarnContext.YARN_SCHEDULER_DRIVER_MEMORY_MB, "123", DriverMemory.class);
+    setConfigs(Keys.corePackageUri(), new File(".").getName(), HeronCorePackageName.class);
+
+    Set<String> reefSpecificConfigs = new HashSet<>();
+    reefSpecificConfigs.add("JobControlHandler");
+    reefSpecificConfigs.add("NodeDescriptorHandler");
+    reefSpecificConfigs.add("ResourceAllocationHandler");
+    reefSpecificConfigs.add("ResourceStatusHandler");
+    reefSpecificConfigs.add("RuntimeStatusHandler");
+    reefSpecificConfigs.add("VerboseLogMode");
+    reefSpecificConfigs.add("HttpPort");
+    reefSpecificConfigs.add("DriverIdentifier");
+
+    Config.Builder builder = new Config.Builder();
+    for (String s : testConfigMap.keySet()) {
+      builder.put(s, testConfigMap.get(s));
+    }
+    builder.put(Keys.stateManagerClass(), "statemanager");
+    builder.put(Keys.packingClass(), "packing");
+    Config config = builder.build();
+
+    spyLauncher.initialize(config, null);
+    Configuration constructedConfig = spyLauncher.getHMDriverConf();
+
+    for (NamedParameterNode<?> reefConfigNode : constructedConfig.getNamedParameters()) {
+      String value = expectedMap.get(reefConfigNode.getName());
+      if (value != null) {
+        Assert.assertEquals(value, constructedConfig.getNamedParameter(reefConfigNode));
+        expectedMap.remove(reefConfigNode.getName());
+      } else {
+        reefSpecificConfigs.remove(reefConfigNode.getName());
+      }
+    }
+    Assert.assertEquals(0, expectedMap.size());
```

Comment:
If either of these assertions fails, the reason won't be clear to the reader of the logs. Would you please add a string reason for the failure (e.g., an unexpected config found) and output the map/set.